src/app/modules/form/public-form/__tests__/public-form.controller.spec.ts

```diff
@@ -389,4 +421,741 @@ describe('public-form.controller', () => {
       expect(mockRes.redirect).toHaveBeenCalledWith(expectedRedirectPath)
     })
   })
+
+  describe('handleGetPublicForm', () => {
+    const MOCK_FORM_ID = new ObjectId().toHexString()
+    const MOCK_USER_ID = new ObjectId().toHexString()
+    const MOCK_USER = {
+      _id: MOCK_USER_ID,
+      email: 'randomrandomtest@example.com',
+    } as IPopulatedUser
+
+    const MOCK_SCRUBBED_FORM = ({
+      _id: MOCK_FORM_ID,
+      title: 'mock title',
+      admin: { _id: MOCK_USER_ID },
+    } as unknown) as PublicForm
+
+    const BASE_FORM = {
+      admin: MOCK_USER,
+      _id: MOCK_FORM_ID,
+      title: MOCK_SCRUBBED_FORM.title,
+      getUniqueMyInfoAttrs: jest.fn().mockReturnValue([MyInfoAttribute.Name]),
+      getPublicView: jest.fn().mockReturnValue(MOCK_SCRUBBED_FORM),
+    }
+
+    const MOCK_REQ = expressHandler.mockRequest({
+      params: {
+        formId: MOCK_FORM_ID,
+      },
+    })
+
+    const MOCK_MYINFO_COOKIE = {
+      accessToken: 'cookie',
+      usedCount: 0,
+      state: MyInfoCookieState.Success,
+    }
+
+    let MOCK_REQ_WITH_COOKIES: ReturnType<typeof expressHandler.mockRequest>
+
+    beforeEach(() => {
+      MOCK_REQ_WITH_COOKIES = expressHandler.mockRequest({
+        params: {
+          formId: MOCK_FORM_ID,
+        },
+        others: { cookies: { MyInfoCookie: MOCK_MYINFO_COOKIE } },
+      })
+    })
+
+    // Success
+    describe('valid form id', () => {
+      const MOCK_JWT_PAYLOAD: JwtPayload = {
+        userName: 'mock',
+        rememberMe: false,
+      }
+
+      beforeAll(() => {
+        MockFormService.setIsIntranetFormAccess.mockImplementation(
+          (_, publicForm) => ok({ ...publicForm, isIntranetUser: false }),
+        )
+
+        MockFormService.setMyInfoError.mockImplementation((publicForm) =>
+          ok({ ...publicForm, myInfoError: false }),
+        )
+      })
+
+      it('should return 200 when there is no AuthType on the request', async () => {
+        // Arrange
+        const MOCK_NIL_AUTH_FORM = (mocked({
+          ...BASE_FORM,
+          authType: AuthType.NIL,
+        }) as unknown) as MockedObject<IPopulatedForm>
+        const MOCK_RES = expressHandler.mockResponse()
+
+        MockAuthService.getFormIfPublic.mockReturnValueOnce(
+          okAsync(MOCK_NIL_AUTH_FORM),
+        )
+        MockFormService.checkFormSubmissionLimitAndDeactivateForm.mockReturnValueOnce(
+          okAsync(MOCK_NIL_AUTH_FORM),
+        )
+
+        // Act
+        await PublicFormController.handleGetPublicForm(
+          MOCK_REQ,
+          MOCK_RES,
+          jest.fn(),
+        )
+
+        // Assert
+        expect(MOCK_RES.json).toHaveBeenCalledWith({
+          form: MOCK_NIL_AUTH_FORM.getPublicView(),
+          isIntranetUser: false,
+          myInfoError: false,
```

Comment:
This is a small change from the original implementation -- when there were no errors, `myInfoError` was set to `undefined` instead of `false`. This means that the frontend will not even receive this key. 

Not sure if we want to send the `myInfoError` key-value if there are no MyInfo errors.

Reply:
i was talking to @mantariksh on this and i think a consistent shape should be enforced.

Reply:
personally I refer returning a consistent shape rather than letting things be possibly undefined. makes no difference to the frontend